core/services/relay/evm/mercury/data_source.go

```diff
@@ -56,41 +56,62 @@ func toBigInt(val interface{}) (*big.Int, error) {
 	return dec.BigInt(), nil
 }
 
-// parse converts the FinalResult into a Observation and stores it in the bridge metadata
-// expects the output of observe to be five values, in the following order:
-// - benchmark price
-// - bid
-// - ask
-// - current block number
-// - current block hash
-func (ds *datasource) parse(result pipeline.FinalResult) (obs relaymercury.Observation, merr error) {
-	if result.HasErrors() {
-		return obs, result.CombinedError()
+// parse expects the output of observe to be five values, in the following order:
+// 1. benchmark price
+// 2. bid
+// 3. ask
+// 4. current block number
+// 5. current block hash
+//
+// returns error on parse errors: if something is the wrong type
+func (ds *datasource) parse(trrs pipeline.TaskRunResults) (obs relaymercury.Observation, merr error) {
+	// pipeline.TaskRunResults comes ordered by index, this is guaranteed by the pipeline executor
+	if len(trrs) != 5 {
+		return obs, fmt.Errorf("invalid number of results, expected: 5, got: %d", len(trrs))
 	}
-	vals := result.Values
-	if len(vals) != 5 {
-		return obs, fmt.Errorf("invalid number of results, got: %s", vals)
-	}
-	for i := 0; i < len(vals); i++ {
-		var err error
+	for i := 0; i < 5; i++ {
```

Comment:
pre-existing, but this ordered loop and select pattern seems odd - aren't we ultimately just running each of those select cases, one at a time, in-order? Why does the loop/select with double indentation add?